app/src/main/java/com/omkarmoghe/pokemap/views/settings/ThemeActivity.java

```diff
@@ -72,34 +72,44 @@ public void onRadioButtonClicked(View v) {
                     editor.putInt(PREF_ID, R.style.AppThemeSquirtle);
                     editor.putInt(PREF_ID_NO_ACTION_BAR, R.style.AppThemeSquirtle_NoActionBar);
                     editor.apply();
+                    setTheme(R.style.AppThemeSquirtle);
+                    recreate();
                 }
                 break;
             case R.id.radioButton2:
                 if(checked) {
                     editor.putInt(PREF_ID, R.style.AppThemeCharmander);
                     editor.putInt(PREF_ID_NO_ACTION_BAR, R.style.AppThemeCharmander_NoActionBar);
                     editor.apply();
+                    setTheme(R.style.AppThemeCharmander);
+                    recreate();
                 }
                 break;
             case R.id.radioButton3:
                 if(checked) {
                     editor.putInt(PREF_ID, R.style.AppThemeBulbasaur);
                     editor.putInt(PREF_ID_NO_ACTION_BAR, R.style.AppThemeBulbasaur_NoActionBar);
                     editor.apply();
+                    setTheme(R.style.AppThemeBulbasaur);
+                    recreate();
                 }
                 break;
             case R.id.radioButton4:
                 if(checked) {
                     editor.putInt(PREF_ID, R.style.AppThemePikachu);
                     editor.putInt(PREF_ID_NO_ACTION_BAR, R.style.AppThemePikachu_NoActionBar);
                     editor.apply();
+                    setTheme(R.style.AppThemePikachu);
+                    recreate();
                 }
                 break;
             case R.id.radioButton5:
                 if(checked) {
                     editor.putInt(PREF_ID, R.style.AppTheme);
                     editor.putInt(PREF_ID_NO_ACTION_BAR, R.style.AppTheme_NoActionBar);
                     editor.apply();
+                    setTheme(R.style.AppTheme);
+                    recreate();
                 }
                 break;
         }
```

Comment:
maybe refactor this in case more themes will be added....
```
int prefId = R.style.AppTheme;
        int prefIdNoAb = R.style.AppTheme_NoActionBar;

        switch(v.getId()) {
            case R.id.radioButton1:
                if(checked) {
                    prefId = R.style.AppThemeSquirtle;
                    prefIdNoAb = R.style.AppThemeSquirtle_NoActionBar;
                }
                break;
            case R.id.radioButton2:
                if(checked) {
                    prefId = R.style.AppThemeCharmander;
                    prefIdNoAb = R.style.AppThemeCharmander_NoActionBar;
                }
                break;
            case R.id.radioButton3:
                if(checked) {
                    prefId = R.style.AppThemeBulbasaur;
                    prefIdNoAb = R.style.AppThemeBulbasaur_NoActionBar;
                }
                break;
            case R.id.radioButton4:
                if(checked) {
                    prefId = R.style.AppThemePikachu;
                    prefIdNoAb = R.style.AppThemePikachu_NoActionBar;
                }
                break;
            case R.id.radioButton5:
            default:
                if(checked) {
                    prefId = R.style.AppTheme;
                    prefIdNoAb = R.style.AppTheme_NoActionBar;
                }
                break;
        }

        editor.putInt(PREF_ID, prefId);
        editor.putInt(PREF_ID_NO_ACTION_BAR, prefIdNoAb);
        editor.apply();
        setTheme(prefId);
        recreate();
```